text/0001-package-structure.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: `package-structure`
+- Start Date: 2024-10-24
+- RFC PR: [CMIP-REF/rfcs#0001](https://github.com/CMIP-REF/rfcs/pull/0001)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes how we should structure the git repository to enable the rapid development of the project while it is immature.
+The core of the project will have some reuse outside of the REF frontend
+so it should be publishable as a separate package.
+The goal is to make it easier for users to extend the project with new metrics providers.
+
+# Motivation
+[motivation]: #motivation
+
+* At the start of this project we will be in a state of flux until we have a working prototype.
+ APIs will change, and we will be adding and removing features.
+* Each metrics package will come with its own set of dependencies. Odds are that they will clash with other packages.
+* We want something that is easy to extend in future, incorporating new metrics providers/metrics.
+* Consistent tooling across the project
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+It is proposed that we start with a mono-repo structure consisting of multiple packages,
+each with their own set of dependencies.
+The tooling and CI will be shared across all projects.
+There will be a `packages` directory at the root of the repo, with each package in its own subdirectory.
+An example has been implemented in [CMIP-REF/cmip-ref#1](https://github.com/CMIP-REF/cmip-ref/pull/0001),
+which contains a `ref-core` package and an example `ref-metrics-example` package.
+
+Each package will have a `README.md` file that describes the purpose of the package and how to use it,
+its own `pyproject.toml` describing the package and it's dependencies, 
+and their own set of tests.
+The test suite for each package is run with only the dependencies required for that package.
+
+A common set of integration tests will also be run to ensure that the packages work together.
+
+The purpose of this structure is to split the project into smaller, more manageable parts
+and to differentiate between the "library" part of the project which is reusable
+and the "application" which builds ontop of the library + metrics packages + additional services.
+
+An additional benefit is that it will be easier to refactor code if it is all accessible in the same repository.
+
+## Packages
+### `ref-core`
+This package will be a library containing the core functionality of the REF.
+This package will be a dependency of all other packages as it will describe the interfaces that metrics providers must implement. 
+This allows us to keep the core functionality separate from the metrics providers.
+
+### `ref-metrics-*`
```

Comment:
It would be easier for people who want to use the REF with a custom metric provider if these were maintained outside of the main REF repository. If we move the example to a separate repository, it will be much easier for relative outsiders to see how to implement the interface because they don't have to dig through all of the REF infrastructure and configuration to find what they're looking for.